app/routers/auth.py

```diff
@@ -0,0 +1,226 @@
+from fastapi import APIRouter, Depends, HTTPException, Security, status
+from fastapi.responses import PlainTextResponse, FileResponse
+from sqlalchemy.orm import Session
+from uuid import UUID
+from typing import Annotated
+from fastapi.security import OAuth2PasswordRequestForm
+from datetime import timedelta
+from app.middleware.auth import get_current_active_user, oauth2_scheme
+from app.schemas.auth import AuthTokenResponse, AuthTokenFamily
+from app.schemas.user import UserFromDB, UserLogin, UserRegister
+from app.schemas.settings import settings
+from app.database import get_db
+import app.services.auth as auth_service
+import app.services.user as user_service
+
+router = APIRouter(
+    prefix="/auth",
+    tags=["authentication"],
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Not found"}
+    },
+)
+
+
+@router.post(
+    "/register",
+    response_model=AuthTokenResponse,
+    description="Create user and login them. Server ignores given scopes."
+)
+async def register(user: UserLogin, db: Session = Depends(get_db)):
+    try:
+        user_db = user_service.register(
+            user=UserRegister.model_validate(user.model_dump()),
+            db=db
+        )
+        return auth_service.login(
+            username=user_db.username,
+            plain_password=user.plaintext_password,
+            db=db
+        )
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e)
+        )
+
+
+@router.post(
+    "/login",
+    response_model=AuthTokenResponse,
+    summary="Get access and refresh tokens"
+)
+async def login(
+    form_data: Annotated[OAuth2PasswordRequestForm, Depends()],
+    db: Session = Depends(get_db)
+):
+    try:
+        return auth_service.login(
+            username=form_data.username,
+            plain_password=form_data.password,
+            scopes=form_data.scopes,
+            db=db
+        )
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e)
+        )
+
+
+@router.post(
+    "/refresh",
+    response_model=AuthTokenResponse,
+)
+async def refresh(
+    refresh_token: str,
+    access_token: str = Depends(oauth2_scheme),
+    db: Session = Depends(get_db),
+    expires_delta: timedelta | None = None,
+):
+    try:
+        return auth_service.refresh(
+            refresh_token,
+            access_token,
+            db,
+            expires_delta,
+        )
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e)
+        )
+
+
+@router.post(
+    "/logout",
+    status_code=status.HTTP_204_NO_CONTENT,
+    summary="Logout user",
+    description="Logout current user. Requires to be logged in.",
+)
+async def logout_user(
```

Comment:
Všude máš register, login, refresh, tak proč tady logout_user?
```suggestion
async def logout(
```